### PR TITLE
Export monitoring : add on option to convert epochdate to date  (#1248)

### PR DESCRIPTION
**Pull request by freddidierRTE**
Release note : 

Feature : 

#1248 : Export monitoring : add on option to convert epochdate to date

### CODE REVIEW

File: ui/main/src/app/common/jsontoarray/json-to-array.spec.ts [hunk 1]
```diff
@@ -140,6 +140,40 @@ describe('flatten simple objects ', () => {
 
   });
 
+  it('flatten objet with epochDate convert to date  ', () => {
```
Comment (AlexGuironnetRTE) [inline ui/main/src/app/common/jsontoarray/json-to-array.spec.ts:143]:
```suggestion
  it('flatten object with epochDate type and valid value => should be converted to date  ', () -> {
```

File: ui/main/src/app/common/jsontoarray/json-to-array.spec.ts [hunk 2]
```diff
@@ -140,6 +140,40 @@ describe('flatten simple objects ', () => {
...
+    const wantedResult = [["column1", "column2"], [new Date(1625838663000), "f2"]];
+
+    const jsonToArray: JsonToArray = new JsonToArray(rules);
+    jsonToArray.add(jsonToConvert);
+
+    expect(areArraysEquals(jsonToArray.getJsonAsArray(), wantedResult)).toEqual(true);
+
+  });
+
+  it('flatten objet with epochDate convert to date but invalid epoch date -> should not convert and keep initial value  ', () => {
```
Comment (AlexGuironnetRTE) [inline ui/main/src/app/common/jsontoarray/json-to-array.spec.ts:160]:
```suggestion
  it('flatten object with epochDate type but invalid value -> should not convert and keep initial value  ', () => {
```

File: services/core/businessconfig/src/main/java/org/opfab/businessconfig/model/MonitoringExportFieldData.java [hunk 3]
```diff
@@ -27,9 +27,9 @@
 @Slf4j
 public class MonitoringExportFieldData implements MonitoringExportField {
   private String columnName;
+  private String jsonField; 
+  private String type;
```
Comment (AlexGuironnetRTE) [inline services/core/businessconfig/src/main/java/org/opfab/businessconfig/model/MonitoringExportFieldData.java:31]: Shouldn't this be an enum?

File: ui/main/src/app/services/export.service.ts [hunk 4]
```diff
@@ -9,18 +9,21 @@
  */
 import * as XLSX from 'xlsx';
 import {saveAs} from 'file-saver-es';
+import {AOA2SheetOpts, JSON2SheetOpts} from 'xlsx';
```
Comment (AlexGuironnetRTE) [inline ui/main/src/app/services/export.service.ts:12]: Sonar is complaning about a duplicate import: https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&open=AXqMB3eqWV0JQmXFQJNh&pullRequest=1432&resolved=false&types=CODE_SMELL

**Review by AlexGuironnetRTE — CHANGES_REQUESTED** [review]
Tests OK but a few questions/remarks.

Also, it might be beyond the scope of this PR but when I tried changing the timezone, the export keeps printing the times in the initial timezone (OpFab default or my computer's timezone, I don't know). It could be confusing.

File: ui/main/src/app/services/export.service.ts (hunk 4 above)
Comment (freddidierRTE) [reply ui/main/src/app/services/export.service.ts:12]: ok done

File: ui/main/src/app/common/jsontoarray/json-to-array.spec.ts (hunk 2 above)
Comment (freddidierRTE) [reply ui/main/src/app/common/jsontoarray/json-to-array.spec.ts:160]: ok done

File: ui/main/src/app/common/jsontoarray/json-to-array.spec.ts (hunk 1 above)
Comment (freddidierRTE) [reply ui/main/src/app/common/jsontoarray/json-to-array.spec.ts:143]: ok done

File: services/core/businessconfig/src/main/java/org/opfab/businessconfig/model/MonitoringExportFieldData.java (hunk 3 above)
Comment (freddidierRTE) [reply services/core/businessconfig/src/main/java/org/opfab/businessconfig/model/MonitoringExportFieldData.java:31]: ok done

**Comment by freddidierRTE** [comment]
Time zone management will be done in another issue

**Comment by freddidierRTE** [comment]
I modify unit test to test that the new field is well inserted

**Pull request merged**